cmd/auth-handler.go

```diff
@@ -104,6 +104,34 @@ func getRequestAuthType(r *http.Request) authType {
 	return authTypeUnknown
 }
 
+func checkRequestAuthType(r *http.Request, bucket, policyAction, region string) APIErrorCode {
```

Comment:
Please add function comment - why `policyAction` is needed for a function named `checkRequestAuthType`? This is doing both authentication and authorisation (enforcing bucket policy for anonymous request).